sapi/fpm/fpm/zlog.c

```diff
@@ -645,7 +645,7 @@ zlog_bool zlog_stream_set_msg_suffix(
 		stream->msg_final_suffix_len = strlen(final_suffix);
 		len = stream->msg_final_suffix_len + 1;
 		if (stream->msg_final_suffix != NULL) {
-			free(stream->msg_suffix);
+			free(stream->msg_final_suffix);
```

Comment:
This is even better catch. It's correct for the only used call of this function and I think it might be potentially reason for #8517 although not 100% sure about it. Anyway we should definitely fix this. It might need to be slightly changed logic (possibly later) as that function has got some logic that if you pass suffix as well as final_suffix, then final suffix is part of allocated suffix (see above) - that's the actual reason of this bug being introduced. However that path is not used so it will always be allocated below.